### PR TITLE
Improve docs and fix deprecation warnings

### PR DESCRIPTION
**Pull request by mrkkrp**
Should hopefully help avoid confusion expressed in #237.

### CODE REVIEW

File: TUTORIAL.md [hunk 1]
```diff
@@ -86,8 +86,12 @@ with this language extension enabled.
 
 There are a few important caveats to mention about this library:
 
-* By default, any non-2XX status code response results in a runtime exception.
-  See the examples of `setRequestIgnoreStatus` below for more information
+* By default, any non-2XX status code response results in a runtime
```
Comment (snoyberg) [inline TUTORIAL.md]: This is obviously a good change, but I'd like to suggest making it slightly better: change this to say that by default status code will not result in runtime exceptions, contrary to previous behavior of the library.

File: TUTORIAL.md [hunk 2]
```diff
@@ -311,7 +315,7 @@ main = do
 
 ## Non-2XX responses
 
-By default, every request that generates a non-2XX response will generate a
+By default, every request that generates a non-2XX response will throw a
```
Comment (snoyberg) [inline TUTORIAL.md]: Similarly here, let's just fix the offending content instead of correcting it later in the document.

File: TUTORIAL.md [hunk 3]
```diff
@@ -507,18 +531,18 @@ main :: IO ()
 main = do
     manager <- newManager tlsManagerSettings
 
-    request <- parseUrl "http://httpbin.org/get"
+    request <- parseRequest_ "http://httpbin.org/get"
```
Comment (snoyberg) [inline TUTORIAL.md]: I think you need either `let request = parseRequest_` or `request <- parseRequest`.

Comment (snoyberg) [inline TUTORIAL.md]: Similar comments apply below

**Comment by mrkkrp** [comment]
I made the requested changes.

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**